kuma/javascript/src/newsletter.jsx

```diff
@@ -167,13 +162,32 @@ export default function Newsletter() {
         }
     }, []);
 
+    if (!showNewsletter) {
+        return null;
+    }
+
+    if (showSuccessfulSubscription) {
+        return (
+            <section className="newsletter-container">
+                <div id="newsletter-thanks" className="newsletter-thanks">
+                    <h2>
+                        {gettext(
+                            'Thanks! Please check your inbox to confirm your subscription.'
+                        )}
+                    </h2>
+                    <p>
+                        {gettext(
+                            'If you haven’t previously confirmed a subscription to a Mozilla - related newsletter you may have to do so. Please check your inbox or your spam filter for an email from us.'
+                        )}
+                    </p>
+                </div>
+            </section>
+        );
+    }
+
     return (
         <section className="newsletter-container">
-            <div
-                id="newsletter-form-container"
-                className={showNewsletter ? 'newsletter' : 'hidden'}
-                aria-hidden={showNewsletter ? false : true}
-            >
+            <div id="newsletter-form-container" className="newsletter">
                 <form
                     ref={newsletterFormRef}
```

Comment:
I still can't see why you need a ref for this? 
Can't it be done with state within the component?

Reply:
Perhaps there *is* a different way to get it. Right now it used to pass the form into `new FormData(newsletterFormRef.current)`

Reply:
We could throw the values into the state, though that'd also mean a re-render on every keystroke. Not expensive here, but avoidable and I don't think it'd add much in terms of clarity of the code (which might be my lack of imagination).

Reply:
Ah, the old controlled vs not controlled component debate. 

> In most cases, we recommend using controlled components to implement forms. 
https://reactjs.org/docs/uncontrolled-components.html

I personally prefer to control inputs unless in some rare cases where the input might come from the browser rather than the keyboard.